src/client/api/types.ts

```diff
@@ -92,6 +92,14 @@ export type PythonApi = {
         cancel?: CancellationToken,
         reInstallAndUpdate?: boolean
     ): Promise<InstallerResponse>;
+    /**
+     * IInstaller
+     */
+    isProductVersionCompatible(
+        product: Product,
+        semVerRequirement: string,
+        resource?: InterpreterUri
```

Comment:
I would change this to `PythonEnvironment` instead of `InterpreterUri`, thats an old approach, and very vague that has caused a lot of issues, We should be more specific about the target interpreter.